Automattic/wp-calypso
# Posts: Drop sites-list usage

This also replaces `site` props (that are partly used for computed attrs such as `slug`) by `siteId` ones plus selectors, down to leaf components level (such as `PostShare`) -- fortunately, none of these are used by any other consumers!

To test:
* Verify that the posts list still works (http://calypso.localhost:3000/posts). Be sure to click around on pretty much all navigational elements to make sure they still work.
* Try for a JP site, with a version < than the minimum JP site (3.3-dev): Verify that a notice appears, and works.

Question:
* Is the `areAllSitesSingleUser` selector reliable?

Possible follow-up work:
* Get rid of the `siteID` prop (capital D). Redundant/confusing crap.
